### DISCUSSION TITLE
configuration issue with c_cpp_properties.json

### DISCUSSION THREAD

**Discussion by the-moog** [microsoft/vscode-cpptools#9602] [General]
In the file c_cpp_properties.json you can have multiple entries in "configurations".  Valid 'name' keys are anything, but "Win32" and "Linux" are used to indicate the platform and selected accordingly by default.
(BTW:  32 bit windows pretty much no longer exists so why Win32??)  But that's not my issue.

I was working in WSL (using WSL/Remote) and realised that it's pickng up settings from the wrong section.  It's picking up path to GCC and Includes from the Win32 section instead of Linux.  So the path format is wrong.  "C:\\...." etc. instead of /mnt/c/.. or /usr/...   I can change it but the work is cross platform, so if I open the project in Windows, rather than WSL it would be wrong again.

Surely there should be a 'WSL' setting as the paths could easily be /mnt/c/... etc or at least use the Linux settings as the OS is Linux (WSL) not Windows.

**Comment by Colengms** [comment]
Hi @the-moog .  The current preferred way to target WSL is to use the `Remote - WSL` extension.  That enables VS Code to establish a remote connection to the WSL instance.  The UI runs locally and a back-end server runs under the WSL instance.

**Reply by the-moog** [reply]
That is what I am using

**Reply by Colengms** [reply]
Hi @the-moog .  You can name a config with any name you choose.  Those are default names.  If you define multiple configs in your `c_cpp_properties.json`, you can choose between them by clicking on the current config name in the bar at the bottom of the window when a C or C++ file is open.

![image](https://user-images.githubusercontent.com/49173979/179847217-285bd284-0159-49ac-9d7f-3921db74c1de.png)

... or by using the `C/C++: Select a Configuration` command.   On a subsequent launch, the C/C++ extension will use the last configuration you had selected.

When I open a empty folder in VS Code in remote WSL, and run `C/C++: Edit Configurations (JSON)`, I see that it creates a `c_cpp_properties.json` containing a configuration titled, `Linux`.  It would appear to properly default to `Linux` via WSL.  Perhaps you started with a `Win32` configuration in your `c_cpp_properties.json` when you first opened it on WSL, which would cause it to remain set to `Win32` even after adding a config titled `Linux`, until you select it.  I believe you will get the behavior you expect if you select your `Linux` as the current configuration.  The extension should also default to using a config titled `Linux` (if it already exists) the first time you open a folder containing a pre-existing `c_cpp_properties.json` file containing a `Linux` config, on WSL.
